tests/operator_test.go

```diff
@@ -526,40 +526,54 @@ var _ = Describe("[Serial][sig-operator]Operator", Serial, func() {
 			return
 		}
 
-		parseImage = func(name, image string) (registry, imagePrefix, version string) {
-			imageRegEx := regexp.MustCompile(fmt.Sprintf("%s%s%s", `^(.*)/(.*)`, name, `([@:].*)?$`))
+		parseImage = func(image string) (registry, imageName, version string) {
+			var getVersion func(matches [][]string) string
+			var imageRegEx *regexp.Regexp
+			var expectedMatchLen int
+
+			if strings.Contains(image, "@sha") {
+				imageRegEx = regexp.MustCompile(`^(.*)/(.*)(@sha.*:)(.*)$`)
+				expectedMatchLen = 5
+				getVersion = func(matches [][]string) string { return matches[0][3] + matches[0][4] }
+			} else {
+				imageRegEx = regexp.MustCompile(`^(.*)/(.*)(:.*)$`)
+				expectedMatchLen = 4
+				getVersion = func(matches [][]string) string { return matches[0][3] }
+			}
+
 			matches := imageRegEx.FindAllStringSubmatch(image, 1)
 			Expect(matches).To(HaveLen(1))
-			Expect(matches[0]).To(HaveLen(4))
+			Expect(matches[0]).To(HaveLen(expectedMatchLen))
```

Comment:
Not sure I see the point of all the parenthesis in the regexps and the check that they all matched...
If the string matched the regexp, all the groups in parentheses got matched too, even if only by the empty string...
But I guess it doesn't hurt anything, just thinking out loud.

Reply:
Yup. This logic existed previously, but in a second though it looks like it's redundant.
Thanks!